swim/memberlist.go

```diff
@@ -232,7 +244,22 @@ func (m *memberlist) GetMembers() (members []Member) {
 // Reincarnate sets the status of the node to Alive and updates the incarnation
 // number. It adds the change to the disseminator as well.
 func (m *memberlist) Reincarnate() []Change {
-	return m.MakeAlive(m.node.address, nowInMillis(m.node.clock))
+	return m.MakeAlive(m.local.Address, nowInMillis(m.node.clock))
+}
+
+// creates a change that reincarnates the local node. It does not change the
+// state of the node but rather reasserts its state by bumping its incarnation
+// number.
+func (m *memberlist) reincarnationChange() Change {
```

Comment:
This method does more than just generating a change: it also bumps the incarnation number.
Maybe rename it to `bumpIncarnation` and call it from
- reincarnate
- PostLocalUpdate?